manifests/init.pp

```diff
@@ -105,7 +97,6 @@
     # use the full version string for comparisons.
     if $::operatingsystem == 'Solaris' and $::operatingsystemmajrelease == '11' {
       # Strip letters from development builds. Unique to Solaris 11 packaging.
-      # Need to pass the regex as strings for Puppet 3 compatibility.
```

Comment:
So does it get us or lose us anything to keep passing as strings? Or is it a wash?

Reply:
It would be better to pass them in as patterns, but I don't want to risk accidentally breaking something due to some mal-formed regex (e.g. like escaping the wrong thing), so I chose to keep them in as strings.